plasma/limbo-priority-mod.md

```diff
@@ -0,0 +1,58 @@
+# Resolving Limbo Transactions via Exit Priority Modification
+
+Design out of conversation with vi, David Knott, Ben Jones, and Eva Beylin. 🍻
+
+---
+
+## TL;DR
+
+This post introduces an alternative to confirmation signatures in MVP that effectively allows any user to set the exit priority of a transaction (position in the priority queue when exiting). 
+
+## Background
+
+Plasma MVP relies on the concept of confirmation signatures to ensure that both parties to a transaction have data availability before it's considered complete. However, it appears that all proposed versions of the confirmation signature scheme are vulnerable to griefing and introduce annoying user experience artifacts. Generally, confirmation signatures operate on the assumption that all transactions are invalid until they are explicitly valid. 
+
+This has motivated the construction of alternative mechanisms which provide the same guarantees. We introduce a design whereby any party may attempt to set the exit priority of a transaction. This means that the transaction would now be processed earlier in the exit priority queue. This attempt is challengeable if the new priority would be greater than the priority of any of the transaction's inputs, if any of the inputs have been double-spent, or if the transaction's new priority would be lower than it's current priority. 
+
+The effect of this design is that an in-flight transaction that may or may not be included in the Plasma chain (a transaction "in limbo") can now be guaranteed to be included. Furthermore, if all of a transactions inputs were created at a time when the chain was valid, then the transaction will be successfully processed before any invalid transactions are processed.
+
+## Construction
+
+A user may attempt to set the priority of a transaction by submitting a signed transaction and a block number to the root chain, subject to a challenge period. This action is bonded, and the priority of deposit ("out of nowhere") transactions cannot be modified. If no successful challenge occurs during the challenge period, then the transaction will have a priority as if it were the added to the end of the given block. 
+
+A priority modification can be challenged in four ways (higher priority = would be exited sooner, lower priority = would be exited later):
+
+1. Someone shows that the specified transaction was included in a block, and that its original priority is higher than the new priority. This prevents someone from decreasing a transaction's priority. 
+2. Someone shows that any of the transaction's inputs would have a lower priority than the transaction. This prevents someone from moving a transaction "earlier" than any of the transaction's outputs.
+3. Someone requests that someone prove some input to the transaction has a higher priority than new priority. This is necessary if blocks are being withheld and a challenge of type (2) is not directly possible.
```

Comment:
Why "prove some" instead of "prove that all"?

Reply:
You know exactly which input(s) to challenge, so you only need one.

Reply:
On source of my confusion here.

Challenger knows that new priority is incorrect since it is higher than priority of one of the inputs. Priority of the input is obvious since it is in the body of the transaction. Ultimately, it can be checked in the contract on priority modification submission. This is why challenger knows if he can challenge. The only missing bit is actual proof of inclusion, since block has been withheld.